src/Mutation/FilteredCollection.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Softonic\GraphQL\Mutation;
+
+use Softonic\GraphQL\Traits\JsonSerializer;
+
+class FilteredCollection implements MutationObject, \JsonSerializable
+{
+    use JsonSerializer;
+
+    protected $arguments = [];
+
+    protected $config;
+
+    private $hasChanged = false;
+
+    public function __construct(array $arguments = [], array $config = [], bool $hasChanged = false)
+    {
+        $this->arguments  = $arguments;
+        $this->config     = $config;
+        $this->hasChanged = $hasChanged;
+    }
+
+    public function __get(string $key)
+    {
+        $items = [];
+
+        foreach ($this->arguments as $argument) {
+
+            $items[] = $argument->{$key};
+        }
+
+        return new Collection($items, $this->config[$key]->children);
+    }
+
+    public function set(array $data)
+    {
+        foreach ($this->arguments as $argument) {
+            if ($argument instanceof Collection) {
+                $argument->set($data);
+            } else {
+                $this->setItemData($argument, $data);
+            }
+        }
+    }
+
+    public function filter(array $filters)
+    {
+        $filteredData = [];
+        if (!empty($this->arguments[0]) && $this->arguments[0] instanceof Collection) {
+            foreach ($this->arguments as $argument) {
+                $argumentCopy = clone $argument;
+
+                $filteredItems = $this->filterItems($argumentCopy->arguments, $filters);
+
+                $argumentCopy->arguments = $filteredItems;
+
+                $filteredData[] = $argumentCopy;
+            }
+        } else {
+            $filteredItems = $this->filterItems($this->arguments, $filters);
+
+            $filteredData = $filteredItems;
+        }
+
+        return new FilteredCollection($filteredData, $this->config);
+    }
+
+    public function hasChanged(): bool
+    {
+        foreach ($this->arguments as $argument) {
```

Comment:
You can improve this setting hasChanged to true when you detect some child has changed (I think once it is change, you cannot go back to unchanged). By the other hand, if it was changed itself, you don't need to check the children. This  will improve the performance.